app/src/components/widgets/charts/testCasesGrowthTrendChart/testCasesGrowthTrendChart.jsx

```diff
@@ -0,0 +1,356 @@
+        columns: [offsets, bars],
+        type: 'bar',
+        order: null,
+        groups: [['offset', 'bar']],
+        onclick: this.onChartClick,
+        color: (c, d) => {
+          let color;
+          switch (d.id) {
+            case 'bar':
+              if (this.isPositive[d.index]) {
+                color = COLORS.COLOR_POSITIVE;
+                break;
+              }
+              color = COLORS.COLOR_NEGATIVE;
+              break;
+            default:
+              color = null;
+              break;
+          }
+          return color;
+        },
+        labels: {
+          format: (v, id, i) => {
+            let step = this.itemData.length < 20 ? 1 : 2;
+            if (this.isTimeLine && this.itemData.length >= 20) {
+              step = 6;
+            }
+
+            if (isPreview || id !== 'bar' || i % step !== 0) {
+              return null;
+            }
+            return this.isPositive[i] ? v : -v;
+          },
+        },
+      },
+      grid: {
+        y: {
+          show: !isPreview,
+        },
+      },
+      axis: {
+        x: {
+          show: !isPreview,
+          type: 'category',
+          categories: this.itemData.map((item) => {
+            let day;
+            if (this.isTimeLine) {
+              day = moment(item.date)
+                .format('dddd')
+                .substring(0, 3);
+              return `${day}, ${item.date}`;
+            }
+            return `#${item.number}`;
+          }),
+          tick: {
+            values: this.isTimeLine
+              ? this.getTimelineAxisTicks(this.itemData.length)
+              : this.getLaunchAxisTicks(this.itemData.length),
+            width: 60,
+            centered: true,
+            inner: true,
+            outer: false,
+            multiline: this.isTimeLine,
+          },
+        },
+        y: {
+          show: !isPreview,
+          padding: {
+            top: 10,
+            bottom: 0,
+          },
+          label: {
+            text: `${intl.formatMessage(messages.cases)}`,
+            position: 'outer-middle',
+          },
+        },
+      },
+      interaction: {
+        enabled: !isPreview,
+      },
+      padding: {
+        top: isPreview ? 0 : 10,
+        left: isPreview ? 0 : 60,
+        right: isPreview ? 0 : 20,
+        bottom: isPreview || !this.isTimeLine ? 0 : 10,
+      },
+      legend: {
+        show: false,
+      },
+      size: {
+        height: this.height,
+      },
+      tooltip: {
+        grouped: true,
+        position: this.getTooltipPosition,
+        contents: this.renderTooltip,
+      },
+    };
+
+    this.setState({
+      isConfigReady: true,
+    });
+  };
+
+  getLaunchAxisTicks = (itemsLength) => {
+    const step = itemsLength > 6 ? Math.round(itemsLength / 12) : 1;
+    const result = [];
+    for (let i = 0; i < itemsLength; i += step) {
+      result.push(i);
+    }
+    return result;
+  };
+
+  getTimelineAxisTicks = (itemsLength) => {
```

Comment:
Please move to `app/src/components/widgets/charts/common/utils/chartRange.js`

Reply:
moved